src/d/actor/d_a_obj_mkiek.cpp

```diff
@@ -37,65 +42,211 @@ static dCcD_SrcSph sph_check_src = {
+    if (phase_state == cPhs_COMPLEATE_e) {
+        phase_state = MoveBGCreate(M_arcname, MKIEK_DZB_MKIEK, NULL, 0x1220);
+        JUT_ASSERT(0xD9, (phase_state == cPhs_COMPLEATE_e) || (phase_state == cPhs_ERROR_e))
+    }
+    return phase_state;
 }
 
 /* 00000754-0000075C       .text Delete__Q210daObjMkiek5Act_cFv */
 BOOL daObjMkiek::Act_c::Delete() {
-    /* Nonmatching */
+    return TRUE;
 }
 
 /* 0000075C-000007B4       .text Mthd_Delete__Q210daObjMkiek5Act_cFv */
 BOOL daObjMkiek::Act_c::Mthd_Delete() {
-    /* Nonmatching */
+    BOOL result = this->MoveBGDelete();
+    if (fpcM_CreateResult(this) != cPhs_STOP_e) {
+        dComIfG_resDelete(&mPhs, M_arcname);
+    }
+    return result;
 }
 
 /* 000007B4-00000848       .text set_mtx__Q210daObjMkiek5Act_cFv */
 void daObjMkiek::Act_c::set_mtx() {
-    /* Nonmatching */
+    mDoMtx_stack_c::transS(current.pos.x, current.pos.y, current.pos.z);
+    mDoMtx_stack_c::ZXYrotM(shape_angle);
+
+    mpModel->setBaseTRMtx(mDoMtx_stack_c::get());
+    mDoMtx_copy(mDoMtx_stack_c::get(), M_tmp_mtx);
+    mpModelV->setBaseTRMtx(mDoMtx_stack_c::get());
 }
 
 /* 00000848-000008A0       .text init_mtx__Q210daObjMkiek5Act_cFv */
 void daObjMkiek::Act_c::init_mtx() {
-    /* Nonmatching */
+    mpModel->setBaseScale(scale);
+    mpModelV->setBaseScale(scale);
+    set_mtx();
 }
 
 /* 000008A0-00000940       .text check__Q210daObjMkiek5Act_cFv */
 void daObjMkiek::Act_c::check() {
-    /* Nonmatching */
+    if (!(!dComIfGp_getDetect().chk_light(&current.pos) && !mSph.ChkTgHit())) {
+        m460++;
+        if (m460 >= 0x14) {
+          fopAcM_orderOtherEventId(this,mDieEventIdx);
+          mState = STATE_1;
+        }
+    } else {
+      m460 = 0;
+  }
+}
+
+inline int daObjMkiek::Act_c::prm_get_sound() const {
+    return daObj::PrmAbstract(this, PRM_SOUND_W, PRM_SOUND_S);
+}
+
+inline int daObjMkiek::Act_c::prm_get_swSave() const {
+    return daObj::PrmAbstract(this, PRM_SWSAVE_W, PRM_SWSAVE_S);
 }
 
 /* 00000940-00000B28       .text demo_wait__Q210daObjMkiek5Act_cFv */
 void daObjMkiek::Act_c::demo_wait() {
-    /* Nonmatching */
+    if (eventInfo.checkCommandDemoAccrpt()) {
+        if (prm_get_sound() == 0) {
+            mDoAud_seStart(JA_SE_READ_RIDDLE_1);
+        }
+        if (m460 < 0x14) {
+            m460++;
+        } else {
+            if (mpBgW->ChkUsed()) {
+                dComIfG_Bgsp()->Release(mpBgW);
+            }
+
+            GXColor color;
+            color.r = tevStr.mColorC0.r;
+            color.g = tevStr.mColorC0.g;
+            color.b = tevStr.mColorC0.b;
+            color.a = tevStr.mColorC0.a;
+            dComIfGp_particle_setProjection(dPa_name::ID_SCENE_819F,
+                &current.pos, &current.angle, NULL,
+                0xFF, NULL,
+                current.roomNo,
+                &tevStr.mColorK0, &color,
+                NULL);
+
+            int prm = prm_get_swSave();
+            dComIfGs_onSwitch(prm, fopAcM_GetHomeRoomNo(this));
```

Comment:
Inline from debug map
```suggestion
            fopAcM_onSwitch(this, prm_get_swSave());
```